kobo-uncaged/main.go

```diff
@@ -492,21 +496,37 @@ func (ku *KoboUncaged) saveCoverImage(contentID string, size image.Point, imgB64
 		imgDir = "koboExtStorage/images-cache"
 	}
 	imgDir = filepath.Join(ku.bkRootDir, imgDir)
-
 	imgID := imgIDFromContentID(contentID)
-
-	for _, cover := range []koboCover{fullCover, libFull, libGrid} {
+	jpegOpts := jpeg.Options{Quality: ku.KuConfig.Thumbnail.JpegQuality}
+
+	var coverEndings []koboCover
+	switch ku.KuConfig.Thumbnail.GenerateLevel {
+	case generateAll:
+		coverEndings = []koboCover{fullCover, libFull, libGrid}
+	case generatePartial:
+		coverEndings = []koboCover{libFull, libGrid}
+	default:
+		coverEndings = nil
```

Comment:
The default case isn't necessary, as an empty slice is nil.

Reply:
I've had the linter complain about a lack of default case in the past. However, just checked it now, and no complaints, so I'll remove this.